midend/expandLookahead.cpp

```diff
@@ -56,7 +56,11 @@ const IR::Expression* DoExpandLookahead::expand(
     } else if (type->is<IR::Type_Bits>() || type->is<IR::Type_Boolean>()) {
         unsigned size = type->width_bits();
         BUG_CHECK(size > 0, "%1%: unexpected size %2%", type, size);
-        auto expression = new IR::Slice(base->clone(), *offset - 1, *offset - size);
+        const IR::Expression* expression =
+                new IR::Slice(base->clone(), *offset - 1, *offset - size);
+        auto tb = type->to<IR::Type_Bits>();
+        if (tb && tb->isSigned)
+            expression = new IR::Cast(tb, expression);
```

Comment:
Why doesn't it need the cast if the type is Boolean?  I guess we have no tests with extracting `bool` fields?  Do we even allow `bool` (as opposed to `bit<1>` in headers?

Reply:
good point, added a new commit to handle this case.

Reply:
If you see section 7.2.7 of P4-16 working spec, you will see the header supports a `bool`.